osu.Game/Database/IHasGuidPrimaryKey.cs

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.ComponentModel.DataAnnotations.Schema;
+using Newtonsoft.Json;
+using Realms;
+
+namespace osu.Game.Database
+{
+    public interface IHasGuidPrimaryKey
+    {
+        [JsonIgnore]
+        [Ignored]
+        public Guid Guid
+        {
+            get => new Guid(ID);
+            set => ID = value.ToString();
+        }
+
+        [JsonIgnore]
+        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
+        [PrimaryKey]
+        string ID { get; set; }
```

Comment:
This is temporary until `Guid` support is officially added in realm (possible the next beta, possible the next official release)

See https://github.com/realm/realm-dotnet/issues/740#issuecomment-756091380